File: rust/src/models/metered_effective_entitlement_value.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{
    metered_entitlement_spec::MeteredEntitlementSpec,
    metered_entitlement_usage::MeteredEntitlementUsage,
};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MeteredEffectiveEntitlementValue {
    pub spec: MeteredEntitlementSpec,

    pub usage: MeteredEntitlementUsage,
}

impl MeteredEffectiveEntitlementValue {
    pub fn new(spec: MeteredEntitlementSpec, usage: MeteredEntitlementUsage) -> Self {
        Self { spec, usage }
    }
}

File: python/meteroid/models/metered_effective_entitlement_value.py
# this file is @generated
import dataclasses

from ..serialization import BaseModel
from .metered_entitlement_spec import MeteredEntitlementSpec
from .metered_entitlement_usage import MeteredEntitlementUsage


@dataclasses.dataclass
class MeteredEffectiveEntitlementValue(BaseModel):
    spec: MeteredEntitlementSpec

    usage: MeteredEntitlementUsage
